### DISCUSSION TITLE
Using Gammu to unlock Nokia 515 with broken keypad?

### DISCUSSION THREAD

**Discussion by friapro** [gammu/gammu#554] [Q&A]
I have a Nokia 515 ([RM-953](https://www.gsmarena.com/nokia_515-5663.php), Series 40) as my daily driver phone. Recently, the keypad was damaged so that I can't access/unlock the phone. I can still turn it on and off (the "power on"/"decline call" button still works), but I'm unable to enter my PIN code or, critically, accept USB transfer to my laptop.

To be specific, I think one key, not sure which one, is stuck in "pressed down" mode, thereby stopping me from pressing any other buttons. I suspect so since, if left turned on, the phone display lights up again and again.

The problem is that all of my phone book entries, old messages and pictures are now trapped in the phone. I thought I had saved the phone book entries on my SIM card, but either I didn't, or they are stored in a way which can't be read on other devices (an Android phone finds nothing on the SIM card). I need to be able to access my phone to at least get the data out of it.

Similarly with the SD card -- it seems Nokia encrypts it somehow, so that it is not readable on my laptop or in Android phones.

When plugged in via USB to my Windows 7 system, both Windows and Nokia PC Suite detects the phone, and the phone display prompts me to accept mass storage or MTP transfer. Of course, I can't do either, so neither Nokia PC Suite nor Windows will allow me to transfer files.

Since I essentially only need to access the phone once, to transfer files from it, my idea is to somehow use gammu's **presskeysequence** command in order to press the menu key, enter my PIN and unlock my phone. From there, I then hope to be able to retrieve my data from it. The question is, however, if that can be done without first letting my laptop access the phone (i.e. by allowing data transfer on the phone, which I obviously can't do).

When the phone is powered on in "charging mode" (as in, all that is displayed on the phone is a battery charging) and connected to my Debian GNU/Linux 10 (buster) system, **gammu-detect -d** gives the [following output (paste)](https://paste.debian.net/1170786/).

Following that, I tried the following for the .gammurc file:

> [gammu]
> device = /dev/ttyACM0
> name = Nokia Nokia_515
> connection = at
 
However, a **gammu identify** gives me the following:

> Error opening device. Unknown, busy or no permissions.

When starting the phone properly (but, again, obviously not unlocking it with a PIN code, since I can't), and connecting it to my laptop, it starts charging. Running **gammu-detect -d** then gives me a fairly different result -- see log [here](https://paste.debian.net/1170787/). As for **lsusb**, it shows the following:

> Bus 001 Device 015: ID 0421:06d4 Nokia Mobile Phones 

I also tried Wammu's phone wizard functionality. Not sure if that is off topic for this forum, but if requested, I can post results from that, too. In short, Wammu fails to find the phone or says I don't have permission.

In short, is there any way I could either send a key sequence to, or access the files on, my Nokia 515 via gammu, while it is still PIN locked?

I have ordered a replacement keypad for it, but I'm unsure if that will fix things.

Thank you in advance!
